### PR TITLE
Moved and rename runend encoding for primitive

### PR DESCRIPTION
**Pull request by joseph-isaacs**
Moved into their own crate and renamed methods to include primitive, in prep for run end bools

### CODE REVIEW

**Review by a10y — APPROVED** [review]

**Comment by robert3005** [comment]
This is a bit funky, you're encoding a logical dtype in this encoding. Though it's not implemented you could do runend on anything that can be compared. We will end up with RunEndBinary as well?

**Comment by joseph-isaacs** [comment]
This is true, I guess the difference is that a runend bool doesn't need a values array.

**Comment by robert3005** [comment]
Right, I thought RunEndBool is the special one and all the other RunEnds can be handled by one implementation

**Comment by robert3005** [comment]
There's a bit of awkwardness that you can produce RunEnd with Boolean dtype and can also have RunEndBool but it's a choice of the compressor/user (we wouldn't probably ever produce RunEnd bool dtype array)

**Comment by joseph-isaacs** [comment]
Are you objecting to the primitive nature. I could move with into a REE by equality or default package?

**Comment by robert3005** [comment]
I think in practice you will never run end compress non primitive values. However, there's no need to restrict this in name and function. RunEndPrimitive encoding is the same as RunEnd encoding. No need to leak the implementation into the name of the it imho

File: encodings/runend/src/primitive/compress.rs [hunk 1]
```diff
@@ -10,15 +10,15 @@ use vortex_dtype::Nullability;
 use vortex_dtype::{match_each_integer_ptype, match_each_native_ptype, NativePType};
 use vortex_error::VortexResult;
 
-pub fn runend_encode(array: &PrimitiveArray) -> (PrimitiveArray, PrimitiveArray) {
```
Comment (joseph-isaacs) [inline encodings/runend/src/primitive/compress.rs]: It was half leaked here.

**Comment by joseph-isaacs** [comment]
But I will try and remove primitive where I can

File: encodings/runend/src/primitive/compress.rs (hunk 1 above)
Comment (robert3005) [reply encodings/runend/src/primitive/compress.rs]: Yes, these are not the generic methods. However, we probably should add one. The entry point for most encoding right now is places like this https://github.com/spiraldb/vortex/blob/develop/vortex-sampling-compressor/src/compressors/runend.rs

Comment (joseph-isaacs) [reply encodings/runend/src/primitive/compress.rs]: Are you saying have a generic fn in encodings/runend or in compressors/runend?

With signature `runend_decode/encode (array) -> array`?

Comment (joseph-isaacs) [reply encodings/runend/src/primitive/compress.rs]: Or would you think that `RunEndArray` has can either be (end,values)  or (end, true) driven?

File: encodings/runend/src/default/compute.rs [hunk 2]
```diff
@@ -8,7 +8,7 @@ use vortex_dtype::match_each_integer_ptype;
 use vortex_error::VortexResult;
 use vortex_scalar::Scalar;
 
-use crate::RunEndArray;
+use crate::default::RunEndArray;
```
Comment (gatesn) [inline encodings/runend/src/default/compute.rs:10]: I'm not a fan of non-descriptive names like "default", was this a relic of a refactor?

**Review by gatesn — CHANGES_REQUESTED** [review]

File: encodings/runend/src/default/compute.rs (hunk 2 above)
Comment (joseph-isaacs) [reply encodings/runend/src/default/compute.rs:10]: Would you prefer its top level or I could go for equality or values?

**Pull request closed without merging**